src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -95,11 +95,6 @@ public void deleteDocumentByStudentKey(String unencryptedRegistrationKey) {
         deleteDocument(Const.SearchIndex.STUDENT, unencryptedRegistrationKey);
     }
 
-    public void createStudentWithoutDocument(StudentAttributes student)
-            throws InvalidParametersException, EntityAlreadyExistsException {
-        createStudent(student, false);
-    }
-
     public void createStudent(StudentAttributes student, boolean hasDocument)
```

Comment:
boolean hasDocument can't be removed. It's been referenced from too many places and it's getting called for both true and false values.
PFA screenshot for call hierarchy of this function.
![screenshot 114](https://user-images.githubusercontent.com/7335736/50241239-022ac280-03ed-11e9-9ee1-18fa1e5e5cfd.png)

Reply:
If the `true` is only from test. It is safe to remove it.

Reply:
Both values (true and false) are getting passed apart from the test cases.
PFA screenshot for detailed information.

createStudentCascade() in StudentsLogic.java calls createStudent() with either of the values (true/false).
![screenshot 117](https://user-images.githubusercontent.com/7335736/50343971-1981c100-054f-11e9-9480-84c87fdcaa8a.png)

Reply:
@akash1994deep, you should not stop at only one method call (especially when it is on the same class). Is the method `createStudentCascadeWithoutDocument` used in production code?

Reply:
I didn't stop at only one method call. It's just for the reference. It provides the information that different values are being passed for the parameter 'hasDocument' and hence can't be removed.

Reply:
How so? If e.g. `createStudentCascadeWithoutDocument` is only the caller of `createStudentCascade` with 2nd parameter being `false`, and `createStudentCascadeWithoutDocument` is only called by test methods, then it's as good as the parameter `hasDocument` is not used because it can be safely replaced with the `true` counterpart (take a look at the issue description again, you seem to miss this part).

Reply:
Ok yeah, I get your point.
I'll check the same.

Reply:
Line 882 of Logic.java calls resetStudentGoogleId() with true as 'hasDocument' value which in turn calls createStudent() and this is apart from the test cases. So, the parameter can't be removed.

Reply:
@akash1994deep your observation does not tell anything. If all production usages of that method use `true` and all non-production usages can be converted to `true`, then it's as good as the parameter can be removed.

Reply:
@akash1994deep just because `createStudentCascadeWithoutDocument` exists in the production package (and therefore qualifies as a "production code") does not mean it is _actually_ used in production. If you trace the caller of that method, you'll find that all callers of that method are test methods, and (now I'm giving a more explicit hint, which I don't normally do) so this method's `hasDocument` parameter qualifies as one of those that can be removed.